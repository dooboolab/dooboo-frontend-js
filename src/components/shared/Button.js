// @flow
import React, { Component } from 'react';
import styled from 'styled-components';
import { WhiteButton, DarkButton } from '../ui/Buttons';

type Props = {
  id?: string,
  white?: boolean,
  imgSrc?: any,
  srcset?: any,
  txt?: string,
  onClick: () => void,
};

type State = {

};

const Text = styled.span`
  font-size: 14px;
  color: rgb(128, 109, 216);
`;

const LogoImg = styled.img`
  position: absolute;
  left: 16px;
  height: 20px;
  width: 20px;
  object-fit: cover
`;

export class Button extends Component<Props, State> {
  static defaultProps: Props = {
    onClick: () => {},
  };

  render() {
    const { white, onClick, imgSrc, txt, srcset } = this.props;
    if (this.props.white) {
      return (
        <WhiteButton
          onClick={() => onClick()}
        >
          {
            this.props.imgSrc
              ? <LogoImg
                src={this.props.imgSrc}
                srcSet={this.props.srcset}
              />
              : null
          }
          <Text>{this.props.txt}</Text>
        </WhiteButton>
      );
    }
    return (
      <DarkButton
        onClick={() => onClick()}
      >
        {
          this.props.imgSrc
            ? <LogoImg
              src={this.props.imgSrc}
              srcSet={this.props.srcset}
            />
            : null
        }
        <Text>{this.props.txt}</Text>
      </DarkButton>
    );
  }
}

export default Button;
