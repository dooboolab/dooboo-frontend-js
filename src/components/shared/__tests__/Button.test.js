import React from 'react';
import { Button } from '../Button';

import renderer from 'react-test-renderer';

// test for the pure component
describe('Button shared component test', () => {
  let tree;
  const component = <Button white={true} txt='Button 2nd test' />;

  it('component and snapshot matches', () => {
    tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('WhiteButton Interaction', () => {
  let count = 1;
  const onClick = () => {
    count++;
  };

  let rendered;
  let instance;
  const component = <Button onClick={onClick}/>;

  beforeAll(() => {
    rendered = renderer.create(component);
    instance = rendered.root;
  });

  it('Simulate onClick', () => {
    const button = instance.find(
      (el: any) => el.type === 'button',
    );
    button.props.onClick();
    expect(count).toBe(2);
  });
});

describe('RedButton Interaction', () => {
  let count = 1;
  const onClick = () => {
    count++;
  };

  let rendered;
  let instance;
  const component = <Button white={true} onClick={onClick}/>;

  beforeAll(() => {
    rendered = renderer.create(component);
    instance = rendered.root;
  });

  it('Simulate onClick', () => {
    const button = instance.find(
      (el: any) => el.type === 'button',
    );
    button.props.onClick();
    expect(count).toBe(2);
  });
});
