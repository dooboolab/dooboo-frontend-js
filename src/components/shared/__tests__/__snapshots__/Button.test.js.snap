// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button shared component test component and snapshot matches 1`] = `
<button
  className="sc-bwzfXH idXafH"
  onClick={[Function]}
>
  <span
    className="sc-htpNat gUfFlb"
  >
    Button 2nd test
  </span>
</button>
`;
